fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -600,6 +621,28 @@ public Response conditionalDelete(@PathParam("type") String type) throws Excepti
             log.exiting(this.getClass().getName(), "delete(String,String)");
         }
     }
+    
+    
+    private long parseIfModifiedSince() {
+        long modifiedSince = -1;
+        try {
+            // Handle RFC_1123 and RFC_850 formats first. 
+            // e.g "Sun, 06 Nov 1994 08:49:37 GMT", "Sunday, 06-Nov-94 08:49:37 GMT", "Sunday, 06-Nov-1994 08:49:37 GMT"
+            // If 2 digits year is used, then means 1940 to 2039.
+            modifiedSince = httpServletRequest.getDateHeader(HEADERNAME_IF_MODIFIED_SINCE);
```

Comment:
so it can be a long? or timestamp?

Reply:
long, the getDateHeader function returns EpochMilli. will add comment to clarify this.

Reply:
I'm not sure that this pattern is exactly what we want. You have two optional segments and, technically, they could both be present and it would still work.

Reply:
done

Reply:
we want to support 2 different patterns with only one formatter, this is the only way so far which works in my tests. I don't need they work together, just need it works if any of them presents.  :( .   PARSER_FORMATTER.parse(httpHeaders.getHeaderString(HEADERNAME_IF_MODIFIED_SINCE), LocalDateTime::from).     and I require to parse as LocalDateTime, let me add test codes to find out what happens if both present, I would guess it will throw exception.

Reply:
added more test codes to cover negative cases.